src/api/Governance.ts

```diff
@@ -210,4 +223,9 @@ export class Governance extends API {
     const result = await this.fetch<ApiResponse<ProposalComment[]>>(`/proposals/${proposal_id}/comments`)
     return result.data
   }
+
+  async getPassedProposals(proposal_type: ProposalType):Promise<DropdownItemProps[]> {
```

Comment:
This api shouldn't know anything about the front end.

If you need change the shape of an api response please do it i front end logic, don't create endpoints to get each formatted data the front-end needs

Reply:
Ok, I get what you are saying.

I created this endpoint because we don't need the whole proposal data in the front end, just a list of title and an id. 
Sending all that info to the frontend seemed a little bit over the top for populating a dropdown.

And since Governance and Draft proposal submission pages both have this kind of dropdown, it seemed worthy to have an endpoint for it.

Maybe we can find another solution that does not return the data formatted as a dropdown item (so it's agnostic to the frontend needs) but still returns a proposal list with just the title and id.

What are your thoughts?

Reply:
maybe something like 

```js
async getProposalsList(proposal_type: ProposalType, proposal_status: ProposalStatus): Promise<Pick<ProposalAttributes, 'id' | 'title'>>{
...
}
```

??

Reply:
What about add and extra parameter to the current api `properties`, so you can use an `omit` function over the result

```text
GET /api/proposal?status=passed&type=poi&properties=id,title

{
    "success": true,
    "data": [
        {
            "id": "...",
            "title": "..."
        }
    ]
}

```

you just need to add and extra prop to `getProposals` on the api

```typescript
export type GetProposalsFilter = {
  // ...
  properties: (keyof ProposalAttributes)[] 
}

Governance.get().getProposals({ type: 'poi', status: 'passed', properties: ['id', title]  })
```